maximum-depth-of-binary-tree/nhistory.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Definition for a binary tree node.
+ * function TreeNode(val, left, right) {
+ *     this.val = (val===undefined ? 0 : val)
+ *     this.left = (left===undefined ? null : left)
+ *     this.right = (right===undefined ? null : right)
+ * }
+ */
+/**
+ * @param {TreeNode} root
+ * @return {number}
+ */
+var maxDepth = function (root) {
+  // Check root is null or not
+  if (root === null) {
+    return 0;
+  }
+  // Make queue array to store root and depth variable
+  let queue = [root];
+  let depth = 0;
+  // Iterate until there is an element inside of queue
+  while (queue.length > 0) {
+    // Record level size to avoid fault size measuring
+    let levelSize = queue.length;
```

Comment:
안녕하세요 세환님!

for 문 안에 node는 const keyword로 선언된 데 반해, queue랑 levelSize는 let로 선언한 이유가 따로 있을까요?

저 같은 경우는 처음에 변수를 선언할 때 먼저 const keyword로 선언하고 이후에 재할당 된다면 선언을 let으로 변경하는데요.
그렇게 행동하는 편이 나중에 코드를 읽을 때 해당 변수가 어떻게 조작될지 예측할 수 있고, 선언 이후에 실수로라도 의도치 않은 재할당을 방지할 수 있는 이득이 있기 때문이에요.

시간 나신다면, 혹시 아래 suggestion 참고해주셨으면 좋겠습니다!

```suggestion
  const queue = [root];
  let depth = 0;
  // Iterate until there is an element inside of queue
  while (queue.length > 0) {
    // Record level size to avoid fault size measuring
   const levelSize = queue.length;
```

Reply:
안녕하세요 Evan님!

네 코맨트 주신대로 queue와 levelSize는 const로 선언하는게 좋은 접근방법인것 같아요.
저도 실제 업무에는 const로 선언하고 시작하는 편인데 알고리즘 풀이할 땐 습관적으로 let을 과도하게 썼던 것 같습니다. 🤦🏻‍♂️

좋은 피드백 감사드려요!